third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2599,8 +2599,10 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
             }
             let mut success = true;
             for (i, arg_ty) in arg_types.iter().enumerate() {
-                let arg_ty = if cand.get_operation().allows_ref_param_for_value() {
-                    // Drop reference type if there is any.
+                let arg_ty = if cand.get_operation().allows_ref_param_for_value()
+                    && self.mode != ExpTranslationMode::Impl
```

Comment:
I see there is also `ExpTranslationMode::TryImplAsSpec`.  Why should we not also exclude that case?

Reply:
Based on my understanding, `ExpTranslationMode::TryImplAsSpec` will translate a move function into a form that can be called inside a spec. In this case, we can drop inference. cc @wrwg